lib/elixir/test/elixir/access_test.exs

```diff
@@ -86,4 +86,25 @@ defmodule AccessTest do
       Access.pop(struct(Sample, []), :name)
     end
   end
+
+  describe "filter/1" do
+    @test_list [1, 2, 3, 4, 5, 6]
+
+    test "filters in get_in" do
+      assert(get_in(@test_list, [Access.filter(&(&1 > 3))]) == [4, 5, 6])
+    end
+
+    test "retains order in get_and_update_in/1" do
+      assert(get_and_update_in(@test_list, [Access.filter(&(&1 == 3 || &1 == 2))], &({&1 * 2, &1})) == {[4, 6], [1, 2, 3, 4, 5, 6]})
+    end
+
+    test "retains order in pop_in/1" do
+      assert(pop_in(@test_list, [Access.filter(&(&1 == 3 || &1 == 2))]) == {[2, 3], [1, 4, 5, 6]})
+    end
+
+    test "chains with other access functions" do
+      mixed_map_and_list = %{foo: Enum.map(@test_list, &(%{value: &1}))}
+      assert(get_in(mixed_map_and_list, [:foo, Access.filter(&(&1 <= 3)), :value]) == [])
```

Comment:
This test is a bit confusing. Maybe it would be better to filter based on `is_tuple/1`?

Reply:
Not sure exactly what you mean. None of the elements of the list are tuples.  `get_and_update_in` returns a tuples of the get value and update value though.  I think using the @test_list module attribute is just less clear,and maybe it would be clearer with an explicit test list in each test.

Reply:
Your filter is comparing a map with `<= 3`. If you want to reject everything, it would be better to have a filter based on the data structure.

Reply:
Sorry, `is_tuple` was wrong, `is_map` is probably better. You do `&(&1 <= 3)` where `&1` is a map which doesn't really make sense, is `%{value: 1} <= 3` the comparison you wanted to make. It would be better if you changed it to a more natural comparison.

Reply:
Yeah, sorry guys it was supposed to be` &1.value <= 3` I'm pretty sure the test should be failing with that mistake though.

Reply:
It doesn't fail because all terms are comparable, even if it doesn't make sense in this case.

Reply:
Oh, it's just wrong all around.  Shouldn't be asserting an empty list.  I'll fix now.

Reply:
When I said it should fail I was thinking that I had the right side of the comparison right `[1, 2, 3]`